docs/setup/upgrade.asciidoc

```diff
@@ -4,29 +4,23 @@
 Depending on the {kib} version you're upgrading from, the upgrade process to 7.0
 varies.
 
-NOTE: {kib} upgrades automatically when starting a new version, as described in
-<<upgrade-migrations, this document>>.
-Although you do not need to manually back up {kib} before upgrading, we recommend
-that you have a backup on hand. You can use
-<<snapshot-repositories, Snapshot and Restore>> to back up {kib}
-data by targeting `.kibana*` indices. If you are using the Reporting plugin,
-you can also target `.reporting*` indices.
-
 [float]
 [[upgrade-before-you-begin]]
 === Before you begin
 
+WARNING: {kib} automatically runs upgrade migrations when required. To roll back to an earlier version in case of an upgrade failure, you **must** have a backup snapshot available. Use <<snapshot-repositories, Snapshot and Restore>> to back up {kib} data by targeting the `.kibana*` indices. For more information see <<upgrade-migrations, upgrade-migrations>>.
+
 Before you upgrade {kib}:
```

Comment:
Thoughts on adding another item here to prevent custom index templates from interfering with .kibana* indices?
i.e., remove any custom index template that might accidentally "cover" `.kibana*` indices.  This can help avoid saved objects migration issues due to custom index templates causing invalid/unexpected mappings.

Reply:
Added the "preventing migration failures" section and linked to it from here.